packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/MultiFrameViewer/FrameCarousel.js

```diff
@@ -49,73 +49,108 @@ export function NextLabel () {
   )
 }
 
-export const StyledInput = styled.input`
+export const StyledFrame = styled.label`
   ${props => props.theme && css`
-    &:checked + img {
-      outline: ${props.theme.global.colors['neutral-4']} solid;
+    input:checked + img {
+      border: ${props.theme.global.colors['neutral-4']} solid;
+    }
+    
+    input:focus + img {
+      outline: 2px solid ${tint(0.5, props.theme.global.colors.brand)};
+    }
+
+    input:hover + img {
+      outline: 2px solid ${tint(0.5, props.theme.global.colors.brand)};
     }
   `}
-  clip: rect(0 0 0 0);
-  overflow: hidden;
-  position: absolute;
+  input {
+    opacity: 0.01;
+    position: absolute;
+  }
+
 `
 
 export const StyledImage = styled.img`
-  height: 3em;
-  width: 3em;
-  margin: 0.5em;
-  float: center;
+  height: 40px;
   object-fit: cover;
   padding: 0;
+  width: 40px;
 `
 
 class FrameCarousel extends React.Component {
   constructor () {
     super()
+    this.handleScroll = this.handleScroll.bind(this)
     this.handlePrevious = this.handlePrevious.bind(this)
     this.handleNext = this.handleNext.bind(this)
+    this.activeLabel = React.createRef()
+    this.frameList = React.createRef()
+  }
+
+  componentDidMount () {
+    this.handleScroll()
+  }
+
+  handleScroll () {
+    const labelIsAboveContainerTop = this.activeLabel.current?.offsetTop <= this.frameList.current?.scrollTop
+    const labelIsBelowContainerBottom = this.activeLabel.current?.offsetTop >= (this.frameList.current?.scrollTop + this.frameList.current?.clientHeight)
+
+    if (labelIsAboveContainerTop || labelIsBelowContainerBottom) {
+      const newContainerTopPosition = this.activeLabel.current.offsetTop - this.activeLabel.current.offsetHeight
+      this.frameList.current.scrollTop = newContainerTopPosition
+    }
+  }
+
+  handleFrameChange (frameIndex) {
+    const { onFrameChange } = this.props
+    onFrameChange(frameIndex)
```

Comment:
I've omitted the callback, and I think it still works because of the CSS changes you noted? I think because the focus is properly updating, but I'm not entirely sure.

Reply:
I have the changes (adding callback to SubjectViewerStore, MultiFrameContainer, and to handleFrameChange in FrameCarousel) commented out locally, can quickly add and commit if I'm mistaken.

Reply:
With the callback, I was seeing it fall behind if I went to the bottom, then looped around with arrow down/right to top, then back to bottom with keys.

Reply:
Quite possible it's not necessary with correct focus handling. I'll have a look.

Reply:
I think if I add the `handleScroll` to `handleFrameChange` it causes the strange mouse behavior you note, but I'm trying to gather more details and specifics to confirm and explain, I'll update with more info.

Reply:
Which actually makes sense (sorta) as it comes back to what's getting focus, as in when clicking or using arrow keys I'm moving focus, but with buttons I'm not changing the focus.

Reply:
Yep. This is now working as it should with the lifecycle method and adding the scroll handler did cause the incorrect behavior with the click interaction. Thanks.